rustls/src/crypto/tls12.rs

```diff
@@ -1,4 +1,65 @@
-pub(crate) fn prf(out: &mut [u8], hmac_key: &dyn super::hmac::Key, label: &[u8], seed: &[u8]) {
+use super::hmac;
+use super::ActiveKeyExchange;
+use crate::error::Error;
+
+use alloc::boxed::Box;
+
+/// An instantiation of the TLS1.2 PRF with a specific, implicit hash function.
+///
+/// See the definition in [RFC5246 section 5](https://www.ietf.org/rfc/rfc5246.txt).
+///
+/// See [`PrfUsingHmac`] as a route to implementing this trait with just
+/// an implementation of [`hmac::Hmac`].
+pub trait Prf: Send + Sync {
+    /// Completes the given key exchange, and then uses the result to compute
+    /// `PRF(secret, label, seed)`, writing the reuslt into `output`.
+    ///
+    /// This can fail only if the key exchange fails.
+    fn for_kx_shared_secret(
+        &self,
+        kx: Box<dyn ActiveKeyExchange>,
+        peer_pub_key: &[u8],
+        label: &[u8],
+        seed: &[u8],
+        output: &mut [u8],
+    ) -> Result<(), Error>;
+
+    /// Computes `PRF(secret, label, seed)`, writing the result into `output`.
+    ///
+    /// Computing the TLS1.2 PRF is infallible.
+    fn for_secret(&self, secret: &[u8], label: &[u8], seed: &[u8], output: &mut [u8]);
+}
+
+/// Implements [`Prf`] using a [`hmac::Hmac`].
+pub struct PrfUsingHmac<'a>(pub &'a dyn hmac::Hmac);
+
+impl<'a> Prf for PrfUsingHmac<'a> {
+    fn for_kx_shared_secret(
+        &self,
+        kx: Box<dyn ActiveKeyExchange>,
+        peer_pub_key: &[u8],
+        label: &[u8],
+        seed: &[u8],
+        output: &mut [u8],
+    ) -> Result<(), Error> {
+        let shared_secret = kx.complete(peer_pub_key)?;
```

Comment:
was this resolved because you like it as-is, or resolved in code but not pushed? 

I don't feel strongly in either case (it's a nit for sure), just want to double check if you did fix it that the fix didn't get lost in the shuffle.

Reply:
Oops, I misread this and fixed the similar issue in tls13.rs, will address this one too